patches/api/0101-Add-support-for-multiple-plugin-folders.patch

```diff
@@ -0,0 +1,295 @@
++                        }
+                     }
+                 }
++            } else {
++                directory.mkdir();
+             }
+         }
+ 
+@@ -220,7 +228,7 @@ public final class SimplePluginManager implements PluginManager {
+ 
+                             server.getLogger().log(
+                                 Level.SEVERE,
+-                                "Could not load '" + file.getPath() + "' in folder '" + directory.getPath() + "'",
++                                "Could not load '" + file.getPath() + "' in folder '" + file.getParentFile().getPath() + "'",
+                                 new UnknownDependencyException(dependency));
+                             break;
+                         }
+@@ -257,7 +265,7 @@ public final class SimplePluginManager implements PluginManager {
+                         loadedPlugins.add(plugin);
+                         continue;
+                     } catch (InvalidPluginException ex) {
+-                        server.getLogger().log(Level.SEVERE, "Could not load '" + file.getPath() + "' in folder '" + directory.getPath() + "'", ex);
++                        server.getLogger().log(Level.SEVERE, "Could not load '" + file.getPath() + "' in folder '" + file.getParentFile().getPath() + "'", ex);
+                     }
+                 }
+             }
+@@ -281,7 +289,7 @@ public final class SimplePluginManager implements PluginManager {
+                             loadedPlugins.add(plugin);
+                             break;
+                         } catch (InvalidPluginException ex) {
+-                            server.getLogger().log(Level.SEVERE, "Could not load '" + file.getPath() + "' in folder '" + directory.getPath() + "'", ex);
++                            server.getLogger().log(Level.SEVERE, "Could not load '" + file.getPath() + "' in folder '" + file.getParentFile().getPath() + "'", ex);
+                         }
+                     }
+                 }
+@@ -294,7 +302,7 @@ public final class SimplePluginManager implements PluginManager {
+                     while (failedPluginIterator.hasNext()) {
+                         File file = failedPluginIterator.next();
+                         failedPluginIterator.remove();
+-                        server.getLogger().log(Level.SEVERE, "Could not load '" + file.getPath() + "' in folder '" + directory.getPath() + "': circular dependency detected");
++                        server.getLogger().log(Level.SEVERE, "Could not load '" + file.getPath() + "' in folder '" + file.getParentFile().getPath() + "': circular dependency detected");
+                     }
+                 }
+             }
+@@ -343,7 +351,8 @@ public final class SimplePluginManager implements PluginManager {
+     }
+ 
+     private void checkUpdate(File file) {
+-        if (updateDirectory == null || !updateDirectory.isDirectory()) {
++        File updateDirectory = new File(file.getParentFile(), Bukkit.getUpdateFolder());
```

Comment:
Create the update directory somewhere and cache it instead of recreating it every time